### PR TITLE
Events section.

### PR DESCRIPTION
**Pull request by obetomuniz**
In favor of issue #206, follows the PR that adds the Events Section to WebComponents.org integrated with the Meetup.com API.

@zenorocha

### CODE REVIEW

File: src/partials/footer.html.eco [hunk 1]
```diff
@@ -23,6 +23,7 @@
 <script src="<%= @site.url %>/vendor/respond/respond.src.js"></script>
 <![endif]-->
 
+<script src="<%= @site.url %>/js/events.js"></script>
```
Comment (zenorocha) [inline src/partials/footer.html.eco]: Should this be loaded in every page? Please move to `src/documents/events/index.html.eco`.

File: src/layouts/events.html.eco [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<%- @partial('header') %>
+<div class="page-wrap">
+<%- @partial('menu') %>
+<%- @content %>
+</div>
+
+<%- @partial('footer') %>
```
Comment (zenorocha) [inline src/layouts/events.html.eco]: Why not using the `home` layout? Seems to be exactly the same as this one.

File: src/files/js/main.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 (function (window, document, undefined) {
 
     'use strict';
+    
+    window.addEventListener('load', function(){
+        new Events();
+    });
```
Comment (zenorocha) [inline src/files/js/main.js]: Please run this on `src/documents/events/index.html.eco`.

File: src/files/js/events.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                        event_url = (eventsOrdered[i].event_url) ? eventsOrdered[i].event_url : "";
+                        virtualCardTpl = virtualCardTpl.replace(/{{url}}/g, event_url);
+                        
+                        eventsSection.innerHTML += virtualCardTpl;
+                    }
+                } else {
+                    eventsSection.innerHTML = '<p class="no-events">No events found.</p>';
+                }
+
+        },
```
Comment (zenorocha) [inline src/files/js/events.js]: Break into smaller functions.

File: src/documents/events/index.html.eco [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        </a>
+        <a class="title" href="{{url}}" target="_blank">
+            <span itemprop="name">{{name}}</span>
+        </a>
+        <div class="description">
+            <p>{{description}}</p>
+            <a class="read-more" href="{{url}}" target="_blank">Read More &gt;</a>
+        </div>
+    </section>
+</script>
```
Comment (zenorocha) [inline src/documents/events/index.html.eco]: What if we use the `<template>` tag? You can include the `webcomponentsjs` polyfill to make sure it's going to work in every browser.

**Comment by Nevraeka** [comment]
Any updates on this one @obetomuniz @zenorocha ?

**Comment by obetomuniz** [comment]
@Nevraeka @zenorocha Was traveling , I'll start working on it. :guitar:

**Comment by Nevraeka** [comment]
@obetomuniz  what is the status of this?

**Comment by obetomuniz** [comment]
Hello @Nevraeka ... It's ready :smile: , @zenorocha can speak more about it.

:dancer:

**Comment by Nevraeka** [comment]
:+1: cool. Thank you @obetomuniz ! I can't wait to add this feature! I'll follow up w/ @zenorocha then update the site

**Comment by Nevraeka** [comment]
@obetomuniz is this complete?

**Comment by obetomuniz** [comment]
Oh yeah @Nevraeka Since the last contact I dont touched on it. Do you have some consideration?

**Comment by Nevraeka** [comment]
@obetomuniz - cool. I just need to know if it has been reviewed & tested across our supported browsers before I add it. If it hasn't I want to make sure the code is still relevant / up-to-date and I can test locally.

**Comment by Nevraeka** [comment]
@obetomuniz also it has been here a while and it seems some conflicts need to be worked out

**Comment by obetomuniz** [comment]
@Nevraeka just fixed the conflicts, go ahead to check if everything is ok now.

**Comment by obetomuniz** [comment]
Damn, one year of old commits, maybe it's time to close this PR and open a new PR just with the code related with Events Section. Thoughts?

**Comment by obetomuniz** [comment]
I just have one important consideration about this line:
https://github.com/webcomponents/webcomponents.github.io/pull/226/files#diff-37d9875b85deb3f587e0555afbda038dR26 

It's related with the current meetup.com API KEY used. It should be changed before deploy to prod, because currently is my API KEY running there, but I have some personal apps using this same API KEY and sadly meetup.com don't give a method to use multiples API KEYs (they have some nice methods, but no one of them is good for us).

Thoughts?

**Comment by Nevraeka** [comment]
@obetomuniz  - Yeah...since it uses the meetups you've added on meetup we should have a separate one.  @zenorocha or @addyosmani - do we have a G+ account or email we can use to create a meetup profile for webcomponents.org?

**Comment by Nevraeka** [comment]
@obetomuniz another issue I see locally is CORS problems. Is that because I am on localhost? It seems like a server side configuration. CORS doesn't seem to work on gh-pages either. Does it make sense to  move this to a build task to prerender the events page prior to deploy? / @zenorocha @addyosmani

**Comment by Nevraeka** [comment]
reviewed it and got a new key for the group. I'll comment ASAP as I test break points & browsers

**Comment by Nevraeka** [comment]
Tweaking some CSS on this one and adding meetup data. Once this is live we should add the ability to have manual events added via manual markdown (these would not show if the dates have past) Thoughts @obetomuniz & @zenorocha ?

**Comment by Nevraeka** [comment]
@obetomuniz - I updated the API key and the fixes I mentioned. if no meetup groups exist, the user experience is pretty poor. Since it is fairly likely that events on meetup.com aren't as frequent as desired ATM - can we add the ability to manually add events as I stated above before merging this? cc @zenorocha

**Comment by Nevraeka** [comment]
cc @zenorocha @obetomuniz

**Comment by Nevraeka** [comment]
@obetomuniz I am closing this . @tjsavage & I are looking into adding an event section to the redesign for webcomponents.org. I'm going to try to use whatever work we can from this and please let us know if you'd like to assist in the redesign. Thanks again!

**Comment by obetomuniz** [comment]
Hey @Nevraeka! It's okay for me... Let me know more about this redesign, I can help with this if needed.

**Comment by Nevraeka** [comment]
@obetomuniz talk with @tjsavage . We really need help contributing so any help is appreciated

**Pull request closed without merging**